uhabits-core/src/jvmMain/java/org/isoron/uhabits/core/ui/screens/habits/list/HabitCardListCache.kt

```diff
@@ -315,9 +318,16 @@ class HabitCardListCache @Inject constructor(
             fromPosition: Int,
             toPosition: Int
         ) {
+            // Workaround for https://github.com/iSoron/uhabits/issues/968
+            val checkedToPosition = if (toPosition > data.habits.size) {
+                logger.error("performMove: $toPosition is strictly higher than ${data.habits.size}")
+                data.habits.size
```

Comment:
I tried to simulate a call with incorrect `toPosition` by modifying `HabitCardListCache:356` to the following:
```kotlin
                if (prevPosition != currentPosition) performMove(
                    habit,
                    prevPosition,
                    currentPosition + 1
                )
```
Unfortunately, it looks like this check was not able to fix the incorrect call. The app still crashed with the following exception:
```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.isoron.uhabits, PID: 6220
    java.lang.IndexOutOfBoundsException: Index: 4, Size: 3
        at java.util.LinkedList.checkPositionIndex(LinkedList.java:560)
        at java.util.LinkedList.add(LinkedList.java:507)
        at org.isoron.uhabits.core.ui.screens.habits.list.HabitCardListCache$RefreshTask.performMove(HabitCardListCache.kt:329)
```
I haven't thought too much about this, but I think we should have `if (toPosition >= data.habits.size) { ... data.habits.size - 1 }`